### PR TITLE
WIP 3426 linux file storage

### PR DESCRIPTION
**Pull request by sblack-usu**
Do not merge, I'm using this to review the code.

### CODE REVIEW

File: hs_core/models.py [hunk 1]
```diff
@@ -3383,12 +3430,18 @@ def can_view(self, request):
...
-    def get_irods_storage(self):
+    def get_storage(self):
         """Return either IrodsStorage or FedStorage."""
-        if self.resource_federation_path:
-            return FedStorage()
-        else:
-            return IrodsStorage()
+
+        # if self.resource.storage_type == StorageCodes.IRODS:
+        return IrodsStorage()
```
Comment (sblack-usu) [inline hs_core/models.py:3437]: Note to come back here once I have looked over the implementation better.

File: config/hydroshare-config.yaml [hunk 2]
```diff
@@ -1,51 +1,46 @@
 ### Local Configuration Variables ###
```
Comment (sblack-usu) [inline config/hydroshare-config.yaml:1]: This needs to be reverted

File: hs_extend/admin.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
```
Comment (sblack-usu) [inline hs_extend/admin.py:1]: what is the hs_extend module?  It seems unrelated.

File: hydroshare/urls.py [hunk 4]
```diff
@@ -138,7 +138,7 @@
     # one out.
```
Comment (sblack-usu) [inline hydroshare/urls.py:138]: This should be reverted

File: hs_core/migrations/0034_manual_migrate_file_paths.py [hunk 5]
```diff
@@ -333,8 +333,8 @@ def resource_path_is_acceptable(resource, path, test_exists=True):
     return folder, base
 
 
-def get_irods_storage(resource):
-    """ Copy of BaseResource.get_irods_storage """
+def get_storage(resource):
```
Comment (sblack-usu) [inline hs_core/migrations/0034_manual_migrate_file_paths.py:336]: Note to review this

File: hs_core/models.py [hunk 6]
```diff
@@ -47,6 +47,50 @@
 from .languages_iso import languages as iso_languages
 
 
+class StorageCodes(object):
+    """
+    Storage codes describe what file storage medium is in use.
+    Storage is a numeric code 1-4:
```
Comment (sblack-usu) [inline hs_core/models.py:53]: What is 4?  Is this a typo?

File: hs_core/models.py [hunk 7]
```diff
@@ -2750,16 +2794,14 @@ class Meta:
...
                                      null=True, blank=True, storage=IrodsStorage())
     fed_resource_file = models.FileField(upload_to=get_path, max_length=4096,
                                          null=True, blank=True, storage=FedStorage())
-
-    # DEPRECATED: utilize resfile.set_storage_path(path) and resfile.storage_path.
-    # fed_resource_file_name_or_path = models.CharField(max_length=255, null=True, blank=True)
-    # DEPRECATED: use native size routine
-    # fed_resource_file_size = models.CharField(max_length=15, null=True, blank=True)
+    # linux_resource_file = models.FileField(upload_to=get_path, max_length=4096,
+    #                                        null=True, blank=True, storage=LinuxStorage())
```
Comment (sblack-usu) [inline hs_core/models.py:2804]: This commented out code seems important for linux storage

File: hs_core/models.py [hunk 8]
```diff
@@ -3346,7 +3388,12 @@ class BaseResource(Page, AbstractResource):
     # means the resource is not locked
     locked_time = models.DateTimeField(null=True, blank=True)
 
-    # this resource_federation_path is added to record where a HydroShare resource is
+    # storage_type = models.IntegerField(choices=StorageCodes.CHOICES,
+    #                                    editable=False,
+    #                                    default=StorageCodes.IRODS)
+
+    # This only applies if storage_type == FEDERATED
+    # The resource_federation_path is added to record where a HydroShare resource is
```
Comment (sblack-usu) [inline hs_core/models.py:3396]: This commented out piece seems important as well

File: hs_linux/storage.py [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+                        os.makedirs(IRODS_PATH)        
+                if( path[0] != "/" ):
+                        path = path[2:] if path[0:2] == "./" else path
+                        return IRODS_PATH + path
+                else:
+                        return path
+        
+        @property
+        def getUniqueTmpPath(self):
+                pass
```
Comment (sblack-usu) [inline hs_linux/storage.py:19]: This will also likely need to be implemented.

File: hs_core/models.py (hunk 1 above)
Comment (sblack-usu) [reply hs_core/models.py:3437]: Linux storage is effectively turned off because this method always returns IrodsStorage.  Some of the commented-out code will have to un-commented to get this going.

Comment (sblack-usu) [reply hs_core/models.py:3437]: The Linux file storage option does not include a migration from irods to linux.  This is something that will have to be implemented if we decide to migrate existing resources from irods to linux.

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4485](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4485)

File: hs_core/migrations/0034_manual_migrate_file_paths.py (hunk 5 above)
Comment (sblack-usu) [reply hs_core/migrations/0034_manual_migrate_file_paths.py:336]: This is a migration, no reason to update this.

File: hs_linux/storage.py [hunk 10]
```diff
@@ -0,0 +1,140 @@
+from django.core.files.storage import  FileSystemStorage
+from hs_linux.models import LinuxAVU
+import os, tarfile, shutil, errno
+from pprint import pprint
+
+IRODS_PATH = "./irods/"
```
Comment (sblack-usu) [inline hs_linux/storage.py:6]: Why is there an IRODS named variable in the LinuxStorage implementation?  This looks like a root path for configuring a linux storage location on disk.

File: hs_core/models.py (hunk 6 above)
Comment (sblack-usu) [reply hs_core/models.py:54]: looks like there was copied from an access control class and not updated completely for storage codes.
